bridge-token-factory/src/lib.rs

```diff
@@ -342,6 +427,242 @@ impl BridgeTokenFactory {
+        FeeBounds {
+            lower_bound,
+            upper_bound,
+        }
+    }
+
+    //this should be added as per: 10% -> 0.1 = 0.1*10^6
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_deposit_fee_percentage(
+        &mut self,
+        token: &String,
+        eth_to_near: U128,
+        eth_to_aurora: U128,
+    ) -> DepositFeePercentage {
+        let _ = validate_eth_address(token.clone());
+        self.deposit_fee_percentage.insert(
+            token,
+            &DepositFeePercentage {
+                eth_to_near,
+                eth_to_aurora,
+            },
+        );
+        DepositFeePercentage {
+            eth_to_near,
+            eth_to_aurora,
+        }
+    }
+
+    // sets fee bounds for withdraw tokens, default bound: None
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_withdraw_fee_bound(
+        &mut self,
+        token: &String,
+        upper_bound: U128,
+        lower_bound: U128,
+    ) -> FeeBounds {
+        let _ = validate_eth_address(token.clone());
+        self.withdraw_fee_bound.insert(
+            token,
+            &FeeBounds {
+                lower_bound,
+                upper_bound,
+            },
+        );
+        FeeBounds {
+            lower_bound,
+            upper_bound,
+        }
+    }
+
+    //this should be added as per: 10% -> 0.1 = 0.1*10^6
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_withdraw_fee_percentage(
+        &mut self,
+        token: &String,
+        near_to_eth: U128,
+        aurora_to_eth: U128,
+    ) -> WithdrawFeePercentage {
+        let _ = validate_eth_address(token.clone());
+        self.withdraw_fee_percentage.insert(
+            token,
+            &WithdrawFeePercentage {
+                near_to_eth,
+                aurora_to_eth,
+            },
+        );
+        WithdrawFeePercentage {
+            near_to_eth,
+            aurora_to_eth,
+        }
+    }
+
+    /*
+    Sets fee percentage for withdraw per silo per token
+    @param: fee_percent:- it should be with 6 decimals, eg: 10% = 0.1*10^6
+            silo_address:- should be an aurora account
+     */
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_withdraw_fee_percentage_for_token_per_silo(
+        &mut self,
+        silo_address: &AccountId,
+        token: &String,
+        fee_percent: U128,
+    ) -> U128 {
+        require!(
+            is_aurora_engine_account(silo_address.clone()),
+            "Not an aurora account"
+        );
+        match self.withdraw_fee_percentage_per_silo.get(silo_address) {
+            Some(_silo_fee) => {
+                self.withdraw_fee_percentage_per_silo
+                    .get(silo_address)
+                    .unwrap()
+                    .token_to_fee_percentage
+                    .insert(token, &fee_percent);
+            }
+            None => {
+                let mut new_map: LookupMap<String, U128> = LookupMap::new(b"t");
```

Comment:
This logic is incorrect, this map has the same key, so it is shared between the tokens.
You can avoid using a sub map just by contacting `"token_id" + ":" + "silo_id"` and using it as a key.

Reply:
We need to have a `default_fee_percentage` for a silo as well, which is only used when there is no explicit fee set for a token wrt that silo. Should I just add this default fee in the map using silo address as key?

Reply:
Yes, you can use just the silo account id as a key

Reply:
cce1ee19211a8a00b1a8567e867eb183c5e24e98